SwiftCompilerSources/Sources/AST/DiagnosticEngine.swift

```diff
@@ -130,6 +129,17 @@ public struct DiagnosticEngine {
     closure()
   }
 
+  // FIXME: Remove this overload once https://github.com/swiftlang/swift/issues/82318 is fixed.
+  public func diagnose(
+    _ id: DiagID,
+    argsArray args: [DiagnosticArgument],
```

Comment:
one more nitpick: we should avoid abbreviations in API names (which include argument labels). How about `arguments: [DiagnosticArgument]`? And no need to "encode" the type (`Array`) into the argument name.

Reply:
Sounds great!